File: news_server/modules/source/infrastructure/getters/tieba.ts
/**
 * 百度贴吧新闻源获取器
 * 从 newsnow/server/sources/tieba.ts 迁移
 */
import type { SourceGetter } from '../../domain/getter/getter'
import { NewsItem } from '../../domain/newsitem/entity'

interface TiebaResponse {
  data: {
    bang_topic: {
      topic_list: {
        topic_id: string
        topic_name: string
        create_time: number
        topic_url: string
      }[]
    }
  }
}

export const tiebaGetter: SourceGetter = async () => {
  const url = 'https://tieba.baidu.com/hottopic/browse/topicList'
  
  const response = await fetch(url)
  if (!response.ok) {
    throw new Error(`Failed to fetch Tieba: ${response.statusText}`)
  }

  const res = (await response.json()) as TiebaResponse
  
  return res.data.bang_topic.topic_list.map((k) =>
    NewsItem.fromEntity({
      id: k.topic_id,
      title: k.topic_name,
      url: k.topic_url,
    })
  )
}

File: modules/source/infrastructure/getters/tieba.go
package getters

import "context"

func (r *Registry) tieba(ctx context.Context) ([]Item, error) {
	var res struct {
		Data struct {
			BangTopic struct {
				TopicList []struct {
					TopicID   string `json:"topic_id"`
					TopicName string `json:"topic_name"`
					TopicURL  string `json:"topic_url"`
				} `json:"topic_list"`
			} `json:"bang_topic"`
		} `json:"data"`
	}
	if err := r.getJSON(ctx, "https://tieba.baidu.com/hottopic/browse/topicList", nil, &res); err != nil {
		return nil, err
	}
	out := make([]Item, 0, len(res.Data.BangTopic.TopicList))
	for _, k := range res.Data.BangTopic.TopicList {
		out = append(out, Item{ID: k.TopicID, Title: k.TopicName, URL: k.TopicURL})
	}
	return out, nil
}
